test/components/views/rooms/EventTile-test.tsx

```diff
@@ -317,11 +316,12 @@ describe("EventTile", () => {
                 });
 
                 const mockCrypto = {
-                    decryptEvent: async (_ev): Promise<IEventDecryptionResult> => {
+                    decryptEvent: async (_ev: MatrixEvent): Promise<IEventDecryptionResult> => {
                         throw new Error("can't decrypt");
                     },
-                } as CryptoBackend;
+                };
 
+                // @ts-ignore `mockCrypto` is not a `CryptoBackend`, which is an internal type in the js-sdk
```

Comment:
This makes development harder as we won't get errors when the types for the provided fields are wrong (it'll tell you to cast to unknown first if sure) and are left to fend for our own, if we're not going to use the types then why have them in the first place.

Reply:
Could use `Parameters<typeof MatrixEvent["attemptDecryption"]>[0]` as an alternative for CryptoBackend

Reply:
>  if we're not going to use the types then why have them in the first place

This is an odd question. We have the type so that we can use it in the *internals* of js-sdk. The fact that a library defines a type in its internals does not necessarily mean that every application using that library can and should be using that type. Indeed, exposing those types makes a library *way harder* to use because it means application authors have to read through way more documentation to find the thing they are actually supposed to use.

The fundamental problem here is that we are poking into the innards of js-sdk in order to make a UTD event. Really, the problem is that we are calling `MatrixEvent.attemptDecryption`, which is an internal method (and yes, it's parameter is an iternal type). But I don't really have any great ideas for avoiding that.

>  It can and will result in tests which end up testing things not conforming to interfaces, providing incorrect return types and the like. 

Mayyyybe? The existing code doesn't actually implement `CryptoBackend`; it just mocks out one function for it. Who is to say that `attemptDecryption` doesn't care about any of the rest of the interface? And once you're mocking a function, its declared interface seems the least of your problems: what it actually does is far more important for whether the test is realistic.

I would strongly argue that having a `@ts-ignore` in test code  is a lesser evil than making `CryptoBackend` part of js-sdk's public interface.

Anyway, I have applied your suggestion, which seems to solve the problem here.